src/Raven.Server/Documents/Sharding/Operations/ShardedBatchOperation.cs

```diff
@@ -50,7 +51,7 @@ public DynamicJsonArray Combine(Dictionary<int, ShardExecutionResult<BlittableJs
                     continue;
                 }
 
-                _command.MarkShardAsComplete(_resultContext, c.ShardNumber);
+                _command.MarkShardAsComplete(_resultContext, c.ShardNumber, HttpRequest.IsFromStudio());
```

Comment:
move it outside of the loop

Reply:
https://github.com/ravendb/ravendb/pull/16433